internal/dag/gatewayapi_processor.go

```diff
@@ -61,6 +63,41 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 	for _, listener := range p.source.gateway.Spec.Listeners {
 
+		var listenerSecret *Secret
+		var err error
+
+		// Check for TLS on the Gateway.
+		if tls := listener.TLS; tls != nil {
+			// Validate the CertificateRef is configured.
+			if tls.CertificateRef == nil {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "SecretNotConfigured",
+					"Spec.VirtualHost.TLS Secret is not configured")
+				continue
+			}
+
+			// Validate the correct protocol is specified.
+			if listener.Protocol != gatewayapi_v1alpha1.HTTPSProtocolType {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "InvalidProtocol",
+					"Spec.VirtualHost.Protocol %q is not valid.")
+				continue
+			}
+
+			// Validate a v1.Secret is referenced which can be kind: secret & group: core.
+			// ref: https://github.com/kubernetes-sigs/gateway-api/pull/562
+			if strings.ToLower(tls.CertificateRef.Kind) != "secret" && strings.ToLower(tls.CertificateRef.Group) != "v1" {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "InvalidReference",
+					"Spec.VirtualHost.TLS Secret must be type v1.Secret")
+				continue
+			}
+
+			listenerSecret, err = p.source.LookupSecret(types.NamespacedName{Name: tls.CertificateRef.Name, Namespace: p.source.gateway.Namespace}, validSecret)
+			if err != nil {
+				p.Errorf(contour_api_v1.ConditionTypeTLSError, "SecretNotValid",
+					"Spec.VirtualHost.TLS Secret %q is invalid: %s", tls.CertificateRef.Name, err)
+				continue
+			}
```

Comment:
might be helpful to move this to a helper method for validating TLS/getting the TLS secret